lib/matplotlib/tests/test_backend_pdf.py

```diff
@@ -296,15 +296,15 @@ def test_pdfpages_fspath():
         pdf.savefig(plt.figure())
 
 
-@image_comparison(['hatching_legend.pdf'])
-def test_hatching_legend():
+@image_comparison(['hatching_legend.pdf'], style='mpl20')
+def test_hatching_legend(text_placeholders):
     """Test for correct hatching on patches in legend"""
-    fig = plt.figure(figsize=(1, 2))
+    fig = plt.figure(figsize=(1.5, 1))
 
     a = Rectangle([0, 0], 0, 0, facecolor="green", hatch="XXXX")
     b = Rectangle([0, 0], 0, 0, facecolor="blue", hatch="XXXX")
 
-    fig.legend([a, b, a, b], ["", "", "", ""])
+    fig.legend([a, b, a, b], ["green", "blue", "green", "blue"])
```

Comment:
Do you know why we put the elements in twice?
![grafik](https://github.com/user-attachments/assets/cf8998b4-3d3c-4cd6-8d63-97d7133f2448)

Reply:
Would also make sense to have different hatches for a and b here?

Reply:
Becuase the bug this test came in with the fix for was related to putting the same artist in the legend twice https://github.com/matplotlib/matplotlib/issues/4469 

This also suggests that the empty string is an important part of the test.

Reply:
Ahh, I've reverted that change then, and added a comment to note down the reason for the empty strings.